src/types.ts

```diff
@@ -149,45 +148,51 @@ async function checkStatus(hash: string, starknetWrapper: StarknetWrapper): Prom
 
 const ACCEPTABLE_STATUSES: TxStatus[] = ["PENDING", "ACCEPTED_ON_L2", "ACCEPTED_ON_L1"];
 export function isTxAccepted(statusObject: StatusObject): boolean {
-    return ACCEPTABLE_STATUSES.includes(statusObject.tx_status);
+    return ACCEPTABLE_STATUSES.includes(statusObject?.tx_status);
```

Comment:
Why did the question marks become necessary suddenly?

Reply:
Inside the `do while()` loop the two functions `isTxAccepted` & `isTxRejected` didn't return when `statusObject` was `undefined`, but removed them now. https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/279/commits/d4bf1b9755dba3abb800cf20914330786858b2ad